### ISSUE TITLE
send home

### ISSUE THREAD

**Issue by Ktkotiy** [newerton/bombcrypto-bot#61]
hi..
is bot can send hero to home ?

**Comment by newerton** [comment]
No, I could add this functionality, but I have to know exactly when to send which heroes and when to send home.

**Comment by Ktkotiy** [comment]
i saw someone did it in his script here : https://github.com/mpcabete/bombcrypto-bot
and also if u need anything or any help from me just tell me

**Comment by farawaykcmm** [comment]
I have an idea of how to implement sending heroes to house, although it would be smart to add an option to configure which house you own since they can hold a different amount of heroes in them.

First the bot checks who is green (ready to work) and verifies whether they are already working or not. Heroes with green bars working are kept working, and if they're resting, they're sent to work.

Next, it has to check if the house is full (if a grey home button is present), if it's not, then go to the next step. If full, just go back to the game.

Now that we know which heroes are not working and that the house is not full, we have to decide which among those not working to send. Usually, it's preferrable to send those with the highest rarity, so you need an image of the name of all rarities (common to super legend). Then, the bot checks whether there's any super legend, if not, then lok for each rarity going down to common. If there's anyone with a higher rarity than common, the bot will send the highest rarity hero. If they're all commons, anyone will do.

Then it needs to check again if the house is full before it can decide whether to keep sending heroes home or quit.

If it's still not full, it can send the next hero with the highest rarity. Or, if all remaining are commons, send anyone. It's better to send any character than leave the home not full.

The only issue with this strategy is a situation where almost all heroes are green so there's not enough people to fill up the home. In which case it wouldn't fill up the house, wasting it's extra stamina recovery for that time being.

But since it's very unlikely that will ever happen even at the super villa, I don't think it's enough of a worry to not use this tactic,

**Comment by Ktkotiy** [comment]
this guy explained it in his git 
-------------------
Send home feature:
How to use it:
Save a screenshot of the heroes you want to be sent home in the directory: /targets/heroes-to-send-home

How it should behave:
It will automatically load the screenshots of the heroes when starting up. After it clicks in the heroes with the green bar to send them to work, it will look if there is any of the heroes that are saved in the directory in the screen. If tit finds one of the heroes, the bot checks if the home button is dark and the work button is not dark. If both these conditions are true, it clicks the home button.

Troubleshooting:
I have not been able to fine adjust it, so here is some problems that may occur, and how to solve them:
The bot should distinguish between the dark, the clear and the gray home buttons.

If the bot says that a hero is working or home, but he is not, that is because the bot is not detecting the dark home button, make the option "home: home_button_trashhold" smaller. You can also replace the image send-home.png in the targets folder.

If the bot is trapped in an loop clicking in an clear home button, he thinks that the clear button is the dark button, make the option home: home_button_trashhold bigger.

The bot should detect the heroes you saved to the directory.

If the bot clicks the wrong heroes, it thinks that another hero is the one you saved the screenshot. Make the option home: hero_trashhold bigger
If it does not detect your heroes, make it smaller. You can also try replacing the screenshot with another part of the hero.

**Comment by farawaykcmm** [comment]
mpcabete's tactic doesn't work well. I've used his bot for a while and even tweaking the threshold, it never sends the proper heroes and often leaves the house not full.

**Comment by Ktkotiy** [comment]
you right his bot didn't even work with me :XD

**Comment by newerton** [comment]
Guys, I'm going to add this Issue as a Feature, I'll see if I can gather more bitcoins to buy a house on Marketplace, when it's available. As I don't have a house, so I can't analyze in detail the process of sending to the house.

The images from common to super legendary, we already have in the repository (https://github.com/newerton/bombcrypto-bot/tree/main/images/themes/default/heroes_types)

**Comment by Ktkotiy** [comment]
i can help you with that if you wanna come anydesk to my pc i already have home

**Comment by farawaykcmm** [comment]
Differences in the regular use vs. the use with a house:

1) When the house is full, the home button in the heroes tab is grey. It's also grey when you don't have a house at all.
When you have a house and it's not full, the home button is the same color as the rest button when the hero is working. Only difference is the text.
When you click home to send a hero home, the color also changes just like the rest button. Same color too.

2) When a hero is sent to the house, it goes to the bottom of the list of heroes, just like sending a hero to work sends him to the top of the list. When you send multiple heroes to the house, they go to the bottom but in the regular order that those heroes would be if none were working or in the house. Example: If I have heroes 4051, 4052 & 4053 and they're normally in that order, if I send 4051 to the house, it goes to the bottom below 4052 & 4053. And if I send 4052 to the house after that, the order will be 4053 (neither in house or working), 4051 (house) and 4052 (house),

**Comment by Ktkotiy** [comment]
is there any chance
